.github/workflows/lintcheck.yml

```diff
@@ -0,0 +1,115 @@
+name: Lintcheck
+
+on: pull_request
+
+env:
+  RUST_BACKTRACE: 1
+  CARGO_INCREMENTAL: 0
+
+concurrency:
+  # For a given workflow, if we push to the same PR, cancel all previous builds on that PR.
+  group: "${{ github.workflow }}-${{ github.event.pull_request.number}}"
+  cancel-in-progress: true
+
+jobs:
+  # Generates `lintcheck-logs/base.json` and stores it in a cache
+  base:
+    runs-on: ubuntu-latest
+
+    outputs:
+      key: ${{ steps.key.outputs.key }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 2
+
+    # HEAD is the generated merge commit `refs/pull/N/merge` between the PR and `master`, `HEAD^`
+    # being the commit from `master` that is the base of the merge
+    - name: Checkout base
+      run: git checkout HEAD^
```

Comment:
Then I guess my previous comment about this was relevant ^^

Reply:
This should hopefully be an easy fix

Reply:
It works because the default checkout of pull request workflows is a merge commit between the PR and the base branch rather than tip of the PR

e.g for this PR https://github.com/rust-lang/rust-clippy/actions/runs/9528672639/job/26266611003#step:2:86 `HEAD` is b237ce2f262a08e589a45cfaa54a34be8650095d which is a merge commit with parents 73c1bfbc5734af7e086fca13ed406889d643de52 (`master` at the time) and feb0671893505272bdad18be595ea2c4dcea2063 (the tip of the PR)

`HEAD^` is the first parent of the merge commit which is `master`, the second to last commit of the PR would be `HEAD^2^`

Reply:
Words don't make commit graphs easy to picture so here's some artwork

![image](https://github.com/rust-lang/rust-clippy/assets/1830331/879cb33b-49bf-4319-baba-99a5a537da51)

Reply:
Ah, so that's how it works. Thanks for the explanation!